pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -736,58 +885,314 @@ def compute_eigvals(*params: TensorLike, **hyperparams) -> TensorLike:
+            &= \text{diag}(
+            \underset{d\text{ times}}{\underbrace{1, \dots, 1}},
+            \underset{(N-d)\text{ times}}{\underbrace{-1, \dots, -1}}
+            ),
+
+        where we denoted the overall dimension as :math:`N=2^n` for :math:`n` qubits and the
+        subspace dimension as :math:`d`, which is called ``dim`` in code.
+        Our first step is to move to a slightly different convention in order to arrive at a
+        *projector* that has ones and zeros, instead of ones and
+        minus ones (note how :math:`G` is not a projector because :math:`G^2=\mathbb{I}_N\neq G`).
+        To do this, we define
+
+        .. math::
+
+            \Pi = \begin{cases}
+                \sum_{j=0}^{d-1} |j\rangle\langle j| & \text{for }\ d\leq \tfrac{N}{2}\\
+                \sum_{j=d}^{N-1} |j\rangle\langle j| & \text{for }\ d> \tfrac{N}{2}
+            \end{cases},
+
+        so that :math:`G=2\Pi -\mathbb{I}_N` for :math:`d\leq \tfrac{N}{2}` and
+        :math:`G=\mathbb{I}_N - 2 \Pi` for :math:`d>\tfrac{N}{2}`. We may summarize
+        this as :math:`G=\sigma (2\Pi -\mathbb{I}_N)` with the sign :math:`\sigma=\pm 1`
+        defined accordingly for the two cases.
+
+        Now we want to decompose :math:`\Pi`, a sum of the computational basis state projectors
+        :math:`|j\rangle\langle j|` onto the :math:`d` consecutive smallest (largest, for
+        :math:`\sigma=-1`) basis states,
+        into projectors that contain a power of two basis state projectors. We will see below that
+        these are exactly the generators of (multi-controlled) phase shift gates, allowing us
+        to implement the decomposition. As we may flip the sign of these individual phase shift
+        gates, we may also use subtraction when decomposing :math:`\Pi`.
+
+        This decomposition of :math:`\Pi` is based on decomposing the integer :math:`d` into sums
+        and differences of powers of two, which is detailed in a separate section below. This
+        decomposition reads
+
+        .. math::
+
+            d = \sum_{i=0}^{n-1} c_i 2^{n-1-i}, \quad c_i\in\{-1, 0, 1\}.
+
+        Now we need to use this decomposition to combine projectors of :math:`2^{n-1-i}`
+        computational basis state projectors into the desired :math:`\Pi`.
+        We start with the scenario :math:`\sigma=+1`, i.e., :math:`d\leq \tfrac{N}{2}`, in which
+        case this decomposition has the form
+
+        .. math::
+
+            \Pi = \sum_{i=0}^{n-1} c_i \sum_{j=s_i}^{e_i} |j\rangle\langle j|,
+
+        where :math:`s_i` and :math:`e_i` are the start and end positions
+        for the :math:`i`\ th contribution to :math:`\Pi`.
+        Besides computing the integer decomposition itself, this decomposition function is mostly
+        about figuring out the right control structure to realize the correct positioning
+        according to :math:`s_i` and :math:`e_i`.
```

Comment:
I think this part is best illustrated with an example as it's not clear from just the text